app/src/main/java/org/simple/clinic/registration/RegistrationApiV1.kt

```diff
@@ -0,0 +1,17 @@
+package org.simple.clinic.registration
+
+import io.reactivex.Single
+import retrofit2.http.Body
+import retrofit2.http.POST
+
+interface RegistrationApiV1 {
+
+  companion object {
```

Comment:
Companion objects are generally discouraged in Kotlin and are left for Java interop. Can we consider moving this to a top-level constant like:
 ```kotlin
private const val API_VERSION = "v1"

interface RegistrationApiV1 {}
```

Reply:
Hey @vinaysshenoy, we also use companion objects for screen keys. See `PatientEntryScreen.KEY` for example. What do you think of them?

Reply:
Btw, the `const` keyword is only allowed in objects or on top levels.

Reply:
Update 2: property initialization isn't allowed in interfaces either. I think this is the reason why we tried using a companion object.

Reply:
I think that we should stop using constants as the key instances and stick to constructing them whenever we need to.

Reasons:
- The key objects are cheap
- If we need to use multiple instances of a screen, then we'll need to construct new instances of the keys instead of using a static constant, which means we'll have to start mixing constants and instances, which is not optimal.

Yes, I want to move this out into a top level const instead of a companion object. I'm also in favour of inlining that constant in the Retrofit URL itself since it's localised to the interface and not references externally.

Reply:
@saket Are we changing this in this PR?

Reply:
@vinaysshenoy sorry I forgot to respond. I think we can review all companion objects in a new PR. Does that sound okay?